### PR TITLE
BUG: Ensure rolling groupby doesn't segfault with center=True

### LINKED ISSUE #35552: BUG: Segmentation fault in RollingGroupby.mean() with center=True and input with odd length
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd

series = pd.Series(range(1, 5))
series.groupby(series).rolling(center=True, window=3).mean()
```

#### Problem description

With an odd-length series and `center=True`, `RollingGroupby.mean()` causes a segfault. If  `center=True` is not provided, or the input series has an even length, then there is no no segfault (example below).
```python
series = pd.Series(range(1, 6))
series.groupby(series).rolling(center=True, window=3).mean()
```
Outputs:
```
1  0              NaN
2  1              NaN
3  2              NaN
4  3              NaN
5  4    1.474760e+250
dtype: float64
```

The series can be grouped in any way, but the example above seemed simplest.

#### Expected Output

Should output a Series.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : d9fff2792bf16178d4e450fe7384244e50635733
python           : 3.7.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.18.0-25-generic
Version          : #26-Ubuntu SMP Mon Jun 24 09:32:08 UTC 2019
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.0
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 49.2.0.post20200714
Cython           : None
pytest           : 6.0.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #35552
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.1.1.rst [hunk 1]
```diff
@@ -16,7 +16,7 @@ Fixed regressions
 ~~~~~~~~~~~~~~~~~
 
 - Fixed regression where :func:`read_csv` would raise a ``ValueError`` when ``pandas.options.mode.use_inf_as_na`` was set to ``True`` (:issue:`35493`).
--
+- Fixed regression in :class:`pandas.core.groupby.RollingGroupby` where a segfault would occur with ``center=True`` and an odd number of values (:issue:`35552`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.1.1.rst:19]: `pandas.core.groupby.RollingGroupby` doesn't exist, so this link won't work (it's in the `window` module). But, we actually don't have this in the API docs, so won't work anyway. I would just do something like `.groupby(..).rolling(..)` to indicate which operation is affected.

**Comment by jreback** [comment]
can you merge master, looks good

**Comment by jreback** [comment]
thanks @mroeschke

**Comment by jorisvandenbossche** [comment]
@MeeseeksDev backport to 1.1.x

**Pull request merged**